cli/internal/env/env.go

```diff
@@ -9,7 +9,55 @@ import (
 	"github.com/vercel/turbo/cli/internal/util"
 )
 
-func getEnvMap() map[string]string {
+// Details contains a map of environment variables broken down by the source
+type Details struct {
+	Explicit EnvironmentVariableMap
+	Prefixed EnvironmentVariableMap
+}
+
+// DetailedMap contains the composite and the detailed maps of environment variables
+// Composite is used as a taskhash input (taskhash.CalculateTaskHash)
+// Details is used to print out a Dry Run Summary
+type DetailedMap struct {
+	Composite EnvironmentVariableMap
+	Details   Details
+}
+
+// EnvironmentVariableMap is a map of env variables and their values
+type EnvironmentVariableMap map[string]string
```

Comment:
Not blocking:

I would move this above `EnvironmentVariableMap` just to aid the readable flow